build.psm1

```diff
@@ -158,10 +158,39 @@ function Start-PSBuild {
     }
 
     # save Git description to file for PowerShell to include in PSVersionTable
-    #git --git-dir="$PSScriptRoot/.git" describe --dirty --abbrev=60 > "$psscriptroot/powershell.version"
+    $powershellVersion = git --git-dir="$PSScriptRoot/.git" describe --abbrev=60 --long
+    $matchVersion = ([regex]::Match($powershellVersion, "^(v.+)-(\d+)-g(.+)")).Groups.Value
+    $null, $ps6MajorVersion, $ps6MinorVersion, $ps6PatchVersion, $ps6LabelVersion = ([regex]::Match($matchVersion[1], "^v(\d+).(\d+).(\d+)-(.+)")).Groups.Value
+    $formattedVersion = "Version: $($matchVersion[1])"
+    if ($($matchVersion[1]) -ne "0") {
+        $formattedVersion += "  Additional commits: $($matchVersion[2])"
+    }
+    $formattedVersion += "  Commit Hash: $($matchVersion[3])"
+    $formattedVersion > "$PSScriptRoot/test/powershell/Host/assets/powershell.version"
+
+    # Generate version constant for $PSVersionTable
+    $template = @"
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by Build.psm1\Start-PSBuild.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+namespace System.Management.Automation
+{
+    internal class GitCommitInfo
+    {
+        internal const string s_GitCommitInfo = "$formattedVersion";
+        internal static readonly SemanticVersion s_psV6Version = new SemanticVersion($ps6MajorVersion, $ps6MinorVersion, $ps6PatchVersion, "$ps6LabelVersion");
+    }
+}
+"@
 
-    # temporary hack to resolve the version issue
-    "v6.0.0-beta.1" > "$psscriptroot/powershell.version"
+    $null = New-Item -Path "$PSScriptRoot\src\System.Management.Automation" -Name "gen" -ItemType Directory -ErrorAction SilentlyContinue
+    Set-Content -Path "$PSScriptRoot\src\System.Management.Automation\gen\PSVersionInfo.generated.cs" -Value $template
```

Comment:
This section should be moved to a separate function (better to make the function invisible from outside the module, but not required) so that `Start-PSBuild` won't keep expanding its size 😄 
And also, move the call to that function after we check/do ResGen, so that the `System.Management.Automation\gen` folder is guaranteed to exist and the `New-Item` line can be removed.